go/enclave/components/batch_registry.go

```diff
@@ -154,15 +157,33 @@ func (br *batchRegistry) BatchesAfter(ctx context.Context, batchSeqNo uint64, up
 	return resultBatches, resultBlocks, nil
 }
 
+func (br *batchRegistry) GetBatchState(ctx context.Context, hash *common.L2BatchHash) (*state.StateDB, error) {
+	batch, err := br.storage.FetchBatch(ctx, *hash)
+	if err != nil {
+		return nil, err
+	}
+	return getBatchStateAtSeq(ctx, br.storage, batch.SeqNo().Uint64())
+}
+
 func (br *batchRegistry) GetBatchStateAtHeight(ctx context.Context, blockNumber *gethrpc.BlockNumber) (*state.StateDB, error) {
 	// We retrieve the batch of interest.
 	batch, err := br.GetBatchAtHeight(ctx, *blockNumber)
 	if err != nil {
 		return nil, err
 	}
 
+	return getBatchStateAtSeq(ctx, br.storage, batch.SeqNo().Uint64())
+}
+
+func getBatchStateAtSeq(ctx context.Context, storage storage.Storage, seqNo uint64) (*state.StateDB, error) {
+	// We retrieve the batch of interest.
+	batch, err := storage.FetchBatchBySeqNo(ctx, seqNo)
```

Comment:
Possible I'm missing something here, but it looks like both the places that call this already had the batch. If it's the batch hash we need rather than the seq number maybe they can pass that in to save the look up?